cli/src/cli_util.rs

```diff
@@ -1302,17 +1303,40 @@ See https://github.com/martinvonz/jj/blob/main/docs/working-copy.md#stale-workin
             let failed_branches = git::export_refs(tx.mut_repo())?;
             print_failed_git_export(ui, &failed_branches)?;
         }
+
+        if let Some(new_commit) = &maybe_new_wc_commit {
+            //TOFIX: Only create a new commit when @ immutable.
+            let new = tx.mut_repo().check_out(
+                self.workspace_id().clone(),
+                &self.settings,
+                &new_commit,
+            )?;
+            maybe_new_wc_commit = Some(new);
+        }
```

Comment:
I managed to create a new commit on every transaction, however i can't call self.check_rewritable now as it crashes for a transaction like jj new for exemple when new commit does not exists in self.repo().

Is there any other way to check a commit is mutable or not ?

Reply:
I think you may have to implement a specialized version of `check_rewritable()` that uses `tx.mut_repo()` instead of the base `ReadonlyRepo`